File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;
        int[] res = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                res[k] = nums[i];
                k++;
            } 
        }
        for (int i = 0; i< res.length; i++) {
            nums[i] =res[i];
        }
        return k;
        
    }
}

File: 27-remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        start, end = 0,len(nums) - 1
        while start <= end:
            if nums[end] == val:
                end -= 1
                continue
            if nums[start] == val:
                nums[start] = nums[end]
                end -= 1
            start += 1

        return end + 1

#class Solution:
#    def removeElement(self, nums: List[int], val: int) -> int:
#        index = 0
#        for i in range(len(nums)):
#            if nums[i] != val:
#                nums[index] = nums[i]
#                index += 1
#        return index